include/aws/common/atomics_msvc.inc

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/* These are implicitly included, but helps with editor highlighting */
+#include <aws/common/atomics.h>
+#include <aws/common/common.h>
+
+#include <intrin.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <Winnt.h>
+
+#if !(defined(_M_IX86) || defined(_M_X64))
+#    error Atomics are not currently supported for non-x86 MSVC platforms
```

Comment:
are we about to just break the arm build? Can we at least have a fallback?

Reply:
The reason I've disabled this is because I can't test it, and having subtly broken atomics is far worse than having atomics that don't build.